src/status_im/ui2/screens/chat/composer/input.cljs

```diff
@@ -192,49 +192,59 @@
-         :on-text-input (partial on-text-input mentionable-users chat-id)}
+  (let [cooldown-enabled?   (rf/sub [:chats/current-chat-cooldown-enabled?])
+        mentionable-users   (rf/sub [:chats/mentionable-users])
+        timeout-id          (reagent/atom nil)
+        last-text-change    (reagent/atom nil)
+        mentions-enabled?   (get @mentions-enabled? chat-id)
+        props               {:style (text-input-style chat-id)
+                             :ref (:text-input-ref refs)
+                             :max-font-size-multiplier 1
+                             :accessibility-label :chat-message-input
+                             :text-align-vertical :center
+                             :multiline true
+                             :editable (not cooldown-enabled?)
+                             :blur-on-submit false
+                             :auto-focus false
+                             :max-length chat.constants/max-text-size
+                             :placeholder-text-color (colors/theme-colors colors/neutral-40
+                                                                          colors/white-opa-30)
+                             :placeholder (if cooldown-enabled?
+                                            (i18n/label :cooldown/text-input-disabled)
+                                            (i18n/label :t/type-a-message))
+                             :default-value (get @input-texts chat-id)
+                             :underline-color-android :transparent
+                             :auto-capitalize :sentences
+                             :auto-correct false
+                             :spell-check false
+                             :on-content-size-change on-content-size-change
+                             :on-selection-change (partial on-selection-change
+                                                           timeout-id
+                                                           last-text-change
+                                                           mentionable-users)
+                             :on-change
+                             (partial on-change
+                                      last-text-change
+                                      timeout-id
+                                      mentionable-users
+                                      refs
+                                      chat-id
+                                      sending-image)
+                             :on-text-input (partial on-text-input
+                                                     mentionable-users
+                                                     chat-id)}
         input-with-mentions (rf/sub [:chat/input-with-mentions])
-        children (fn []
-                   (if mentions-enabled?
-                     (map-indexed
-                      (fn [index [_ text]]
-                        ^{:key (str index "_" type "_" text)}
```

Comment:
`type` here refers to the clojure function to get the type of something.  So I replaced that `_` with `text-type` and used that var toc reate the react key.

Reply:
Good catch! The original code looks incorrect. Here's the key I got with `(str 0 "_" type "_" "hello")`. Also the condition `(= type :mention)` was always returning false, since `type` is the core function as you said because it wasn't shadowed by a local binding.

```clojure
"0_function cljs$core$type(x){\nif((x == null)){\nreturn null;\n} else {\nreturn x.constructor;\n}\n}_hello"
```

In PRs we usually recommend against bindings that shadow core Clojure functions, one of the reasons is exactly to avoid the problem in the original code you fixed. Common ones I've seen in `status-mobile` PRs are `type` and `count`, but probably others I can't recall now.